File: templates/Features/DragAndDrop.CodeBehind/Services/DragAndDrop/ListViewDropConfiguration.cs
using System;
using Windows.UI.Xaml;

using Param_ItemNamespace.Models;

namespace Param_ItemNamespace.Services.DragAndDrop
{
    public class ListViewDropConfiguration : DropConfiguration
    {
        public static readonly DependencyProperty DragItemsStartingActionProperty =
            DependencyProperty.Register("DragItemsStartingAction", typeof(Action<DragDropStartingData>), typeof(DropConfiguration), new PropertyMetadata(null));

        public static readonly DependencyProperty DragItemsCompletedActionProperty =
            DependencyProperty.Register("DragItemsCompletedAction", typeof(Action<DragDropCompletedData>), typeof(DropConfiguration), new PropertyMetadata(null));

        public Action<DragDropStartingData> DragItemsStartingAction
        {
            get { return (Action<DragDropStartingData>)GetValue(DragItemsStartingActionProperty); }
            set { SetValue(DragItemsStartingActionProperty, value); }
        }

        public Action<DragDropCompletedData> DragItemsCompletedAction
        {
            get { return (Action<DragDropCompletedData>)GetValue(DragItemsCompletedActionProperty); }
            set { SetValue(DragItemsCompletedActionProperty, value); }
        }
    }
}

File: templates/Features/DragAndDrop.CodeBehindVB/Services/DragAndDrop/ListViewDropConfiguration.vb
Imports Param_ItemNamespace.Models

Namespace Services.DragAndDrop

    Public Class ListViewDropConfiguration
        Inherits DropConfiguration

        Public Shared ReadOnly DragItemsStartingActionProperty As DependencyProperty = DependencyProperty.Register("DragItemsStartingAction", GetType(Action(Of DragDropStartingData)), GetType(DropConfiguration), New PropertyMetadata(Nothing))

        Public Shared ReadOnly DragItemsCompletedActionProperty As DependencyProperty = DependencyProperty.Register("DragItemsCompletedAction", GetType(Action(Of DragDropCompletedData)), GetType(DropConfiguration), New PropertyMetadata(Nothing))

        Public Property DragItemsStartingAction As Action(Of DragDropStartingData)
            Get
                Return CType(GetValue(DragItemsStartingActionProperty), Action(Of DragDropStartingData))
            End Get

            Set(value As Action(Of DragDropStartingData))
                SetValue(DragItemsStartingActionProperty, value)
            End Set
        End Property

        Public Property DragItemsCompletedAction As Action(Of DragDropCompletedData)
            Get
                Return CType(GetValue(DragItemsCompletedActionProperty), Action(Of DragDropCompletedData))
            End Get

            Set(value As Action(Of DragDropCompletedData))
                SetValue(DragItemsCompletedActionProperty, value)
            End Set
        End Property
    End Class
End Namespace
